app/models/github/issue_label.rb

```diff
@@ -3,4 +3,20 @@ class Github::IssueLabel < ApplicationRecord
     inverse_of: :labels,
     foreign_key: "github_issue_id",
     class_name: "Github::Issue"
+
+  %w[action knowledge module size status type].each do |label|
+    define_singleton_method "for_#{label}" do |val|
+      return unless const_get(label.pluralize.upcase.to_sym).include?(val)
+
+      "x:#{label}/#{val.to_s.tr('_', '-')}"
+    end
+  end
```

Comment:
I think this all gets easier if this is one method `for_label(type, value)` rather than lots of individual methods. The consumer will find this hard to work with as you've seen in the command. The constants below could then (potentially) be one hash, rather than multiple constants (I've not read through the whole PR yet, so that might not be a good idea)

Reply:
Yep, much better. I think I've left my go a bit overboard with the metaprogramming stuff :)